### PR TITLE
Remove screen -> screen imports in Inventories and Hosts

### PR DESCRIPTION
**Pull request by marshmalien**
##### SUMMARY
Issue: #6141

Bug was due to this line `const { id: inventoryId, hostId: inventoryHostId } = useParams();` in HostDetail.jsx. When navigating from `/hosts`, the :id is the host's id, not the inventory id. This is why the inventory link id matched the host id. 

To fix the issue, I made the components (Detail, Add, Edit views) more deterministic by passing them everything they needed to render through props instead of through router hooks like `useRouteMatch`. 

Also in this PR:
* Move HostForm out into generic components directory
  * Hide/Show Inventory lookup with `isInventoryVisible` flag (defaults to true)
* Duplicate screens where we were importing Host screens into Inventory screens. The example below illustrates the bad pattern we are trying to move away from: 
```
<InventoryHosts>
    <HostAdd />
</InventoryHosts>
```

##### ISSUE TYPE

 - Bugfix Pull Request

##### COMPONENT NAME

 - UI

##### ADDITIONAL INFO
To stay ahead of the curve, React Router docs suggest to start using `<Route children>` instead of using `<Route component>` and `<Route render>` props. In this PR, I've updated some of our route configurations to render regular children instead of using the `render` prop. This will hopefully ease the transition to react-router v6. 

```
// before
<Route path="/hosts/:hostId"
  render={() => (	
    <Host setBreadcrumb={setBreadcrumb} inventory={inventory} />	
  )}	
/>

// after
<Route path="/hosts/:hostId">
    <Host setBreadcrumb={setBreadcrumb} inventory={inventory} />
</Route>
```

Guide: https://github.com/ReactTraining/react-router/blob/dev/docs/advanced-guides/migrating-5-to-6.md#upgrade-to-react-router-v51

Blog: https://reacttraining.com/blog/react-router-v5-1/

### CODE REVIEW

File: awx/ui_next/src/screens/Inventory/InventoryHostDetail/InventoryHostDetail.test.jsx [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+    test('expected api call is made for delete', async () => {
+      await act(async () => {
+        wrapper.find('DeleteButton').invoke('onConfirm')();
+      });
+      expect(HostsAPI.destroy).toHaveBeenCalledTimes(1);
+    });
+
+    test('Error dialog shown for failed deletion', async () => {
+      HostsAPI.destroy.mockImplementationOnce(() =>
```
Comment (AlexSCorey) [inline awx/ui_next/src/screens/Inventory/InventoryHostDetail/InventoryHostDetail.test.jsx:50]: seems like there are about 4 ways we are throwing errors in tests these days.

File: awx/ui_next/src/screens/Inventory/InventoryHostDetail/InventoryHostDetail.test.jsx [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+      readOnlyHost.summary_fields.user_capabilities.edit = false;
+
+      wrapper = mountWithContexts(<InventoryHostDetail host={mockHost} />);
+    });
+
+    afterAll(() => {
+      wrapper.unmount();
+    });
+
+    test('should hide edit button for users without edit permission', async () => {
```
Comment (AlexSCorey) [inline awx/ui_next/src/screens/Inventory/InventoryHostDetail/InventoryHostDetail.test.jsx:84]: 👍

File: awx/ui_next/src/screens/Inventory/InventoryHosts/InventoryHosts.jsx [hunk 3]
```diff
@@ -1,28 +1,22 @@
...
     <Switch>
       <Route key="host-add" path="/inventories/inventory/:id/hosts/add">
-        <HostAdd />
+        <InventoryHostAdd inventory={inventory} />
+      </Route>
+      <Route key="host" path="/inventories/inventory/:id/hosts/:hostId">
+        <InventoryHost setBreadcrumb={setBreadcrumb} inventory={inventory} />
+      </Route>
+      <Route key="host-list" path="/inventories/inventory/:id/hosts">
+        <InventoryHostList />
```
Comment (AlexSCorey) [inline awx/ui_next/src/screens/Inventory/InventoryHosts/InventoryHosts.jsx:18]: We probably want to add a Not Found route to catch erroneous urls.

Comment (marshmalien) [reply awx/ui_next/src/screens/Inventory/InventoryHosts/InventoryHosts.jsx:18]: It looks like we have a Route that catches erroneous urls in the Inventories.jsx file and InventoryHost.jsx file. Maybe I missed a case, what erroneous url would you expect to catch in this component?

If a user navigates to `/inventories/inventory/1/hosts/1/foobar`, then the InventoryHost.jsx file should catch the route. 

If a user navigates to `/inventories/inventory/1/foobar`, then the Inventory.jsx file should catch the route.

File: awx/ui_next/src/screens/Inventory/InventoryHostDetail/InventoryHostDetail.test.jsx (hunk 1 above)
Comment (marshmalien) [reply awx/ui_next/src/screens/Inventory/InventoryHostDetail/InventoryHostDetail.test.jsx:50]: Yeah, `.mockRejectedValue();` is sugar for `.mockImplementation(() => Promise.reject(value));`.

**Review by jlmitch5 — APPROVED** [review]
looks good!

File: awx/ui_next/src/screens/Inventory/InventoryHosts/InventoryHosts.jsx (hunk 3 above)
Comment (AlexSCorey) [reply awx/ui_next/src/screens/Inventory/InventoryHosts/InventoryHosts.jsx:18]: Ahh that's true.  Sounds good to me.

**Review by AlexSCorey — APPROVED** [review]
You addressed all of my concerns offline.

**Pull request merged**